frame/balances/src/lib.rs

```diff
@@ -1831,7 +1828,7 @@ where
 		//   account is attempted to be illegally destroyed.
 
 		for attempt in 0..2 {
-			match Self::mutate_account(who, |account| {
```

Comment:
This code is a bit weird with or without this PR.. could use a second pair of eyes.

Reply:
I've taken a look at this code here https://github.com/paritytech/substrate/pull/12108 . But those code cleanup changes should be separate to this PR.